### PR TITLE
Use bundler to make Rails work

### PR DESCRIPTION
**Pull request by pyama86**
Thanks for contributing to the buildpack. To speed up the process of reviewing your pull request please provide us with:

* A short explanation of the proposed change:

Implements the minimum operation of bundler.
This PR is influenced by [composer-cnb](https://github.com/cloudfoundry/php-composer-cnb).
* An explanation of the use cases your change solves

* [ ] I have viewed signed and have submitted the Contributor License Agreement

* [ ] I have made this pull request to the `develop` branch

* [ ] I have added an integration test

### CODE REVIEW

**Comment by pyama86** [comment]
> All pull request submitters and commit authors must have a Contributor License Agreement (CLA). Click here for details on the CLA process.

oh...PDF...

**Comment by dfreilich** [comment]
Hey @pyama86 , thanks for the PR! We really appreciate the help! I noticed that you weren't a fan of the CLA, so I closed and reopened the issue to trigger our new CLA process, which can be done without a PDF. Once you sign that, we'll be happy to take a look, and hopefully merge it in :)

**Comment by pyama86** [comment]
Hi @dfreilich 
Thank you for the change from CLA PDF signature to signature system.
But it doesn't seem to work.
I think the CSS is probably wrong.

image

```
OS: Windows 10
Browser: 77.0.3865.120（Official Build） （64bit）
```

> and hopefully merge it in

Thank you, when all my work is done, remove WIP and mention.

Thanks！

**Comment by dfreilich** [comment]
@pyama86 If you're having trouble with the EasyCLA system, please file a support ticket (the link is above). I think that we need your confirmation before we can use the code.

**Comment by pyama86** [comment]
@dfreilich I was able to sign!
Thanks!

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+INFO_COLOR=\033[1;34m
+RESET=\033[0m
+BOLD=\033[1m
+TEST ?= $(shell go list ./... | grep -v -e vendor -e keys -e tmp)
+GO ?= GO111MODULE=on $(SYSTEM) go
+
+test: ## Run test
+	@echo "$(INFO_COLOR)==> $(RESET)$(BOLD)Testing$(RESET)"
+	$(GO) test -v $(TEST) -timeout=30s -parallel=4
+	$(GO) test -race $(TEST)
```
Comment (Zanadar) [inline Makefile]: As a general rule we like to provide/use scripts for testing, etc. in a centralized place (https://github.com/cloudfoundry/cnb-tools) but that said, these scripts would be open to an improvement or even some sort of local override system.

File: runner/runner.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	if r.Err != nil {
+		cmd.Stderr = io.MultiWriter(os.Stderr, r.Err)
+	} else {
+		cmd.Stderr = os.Stderr
+	}
+
+	err := cmd.Run()
+	// this is on purpose, we return whatever is in the buffer regardless of an error occurring
+	//  this defers handling of the error to the caller, see CheckPlatformReqs in composer.go
+	return buf.String(), err
```
Comment (Zanadar) [inline runner/runner.go]: I like this pattern

**Comment by pyama86** [comment]
@Zanadar Thank you for your reply. This PR is work in progress. Remove WIP if it works well in my rails apps. 
I am very grateful for the kindness of the people in your company.

**Comment by pyama86** [comment]
@Zanadar @dfreilich 
The implementation I needed is done and the Rails app is working.

This cnb only installs bundler and gem packges, and bundler is installed in the dependent ruby and launch command is supposed to use `procfile-cnb`.

I'm going to maintain it long time, so I'd be happy if you could consider merging.

File: cmd/detect/main.go [hunk 3]
```diff
@@ -31,6 +34,16 @@ func main() {
 // TODO: implement the following
 // 		- install nodjs, npm, and yarn if needed
 func runDetect(context detect.Detect) (int, error) {
+	// TODO: include to image of cloudfoundry:cnb
+	_, err := exec.LookPath("bundle")
+	if err != nil {
+		_, err := exec.Command("gem", "install", "bundler", "-v", "1.17.3", "--user-install").CombinedOutput()
```
Comment (ryanmoran) [inline cmd/detect/main.go]: What happens in the case that this environment is not connected to the internet? Wouldn't this just always fail?

**Comment by dwillist** [comment]
Hey @pyama86 apologies it took us so long to get to this. We have migrated work being done on CNB's [here](https://github.com/paketo-buildpacks) & are working to make this project truly 'open-source'. 

During the interim we have changed the the technical direction of this project & are currently working on adding bundler & rails support using our new framework.

**Comment by pyama86** [comment]
Hi! thank you for sharing!
I understand.

**Pull request closed without merging**